blockchain/state/sync_test.go

```diff
@@ -156,6 +165,48 @@ func checkStateConsistency(db database.DBManager, root common.Hash) error {
 	return it.Error
 }
 
+// compareStatesConsistency checks that all data of given two states.
+func compareStatesConsistency(oldDB database.DBManager, newDB database.DBManager, root common.Hash) error {
+	// Create and iterate a state trie rooted in a sub-node
+	if _, err := oldDB.ReadStateTrieNode(root.Bytes()); err != nil {
+		return err
+	}
```

Comment:
Are they necessary? The existence of root node could be checked with `New` function.

Reply:
Good. :) It is already checked in `oldState, err := New(root, NewDatabase(oldDB))`. 
I remove them.